src/validation.h

```diff
@@ -152,12 +152,20 @@ extern const std::vector<std::string> CHECKLEVEL_DOC;
 FILE* OpenBlockFile(const FlatFilePos &pos, bool fReadOnly = false);
 /** Translation to a filesystem path */
 fs::path GetBlockPosFilename(const FlatFilePos &pos);
+/** Import blocks from one or more external files */
+void LoadExternalBlockFiles(
+    const std::string& action,
+    const CChainParams& chainparams,
+    const std::vector<fs::path>& blk_paths,
+    bool write_to_disk);
 /** Import blocks from an external file */
 void LoadExternalBlockFile(
     const CChainParams& chainparams,
-    FILE* fileIn,
-    FlatFilePos* dbp = nullptr,
-    std::multimap<uint256, FlatFilePos>* blocks_with_unknown_parent = nullptr);
+    const std::vector<fs::path>& blk_paths,
+    size_t n_file,
```

Comment:
Given the name `LoadExternalBlockFile` I'm confused why `blk_paths` plural is passed as a param, instead of just a single file/path

Reply:
That function, `LoadExternalBlockFile()`, does (attempt to) load the blocks from exactly one block file, _but_, doing so may cause, as a side effect, the loading of blocks seen earlier but that could not be loaded then because their parent was missing (and is now present, see my earlier comment). This requires reaching back and reading specific blocks from earlier files, and that requires the full list of block file paths (the `blk_paths` argument). This basic flow is the same as before this PR. I agree this is somewhat confusing, maybe a comment would help. But maybe not, because perhaps there's no shortcut to just studying the code, I'm not sure.